SecretSanta/test/SecretSanta.Business.Tests/GiftServiceTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SecretSanta.Data;
+using SecretSanta.Data.Tests;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SecretSanta.Business.Tests
+{
+
+    [TestClass]
+    public class GiftServiceTests : TestBase
+    {
+        [TestMethod]
+        public async Task InsertAsync_CylonDetector_Success()
+        {
+            // Arrange
+            using var dbContextInsert = new ApplicationDbContext(Options);
+            IGiftService service = new GiftService(dbContextInsert, Mapper);
+
+            var cylonDetector = SampleData.CreateCylonDetector();
+            cylonDetector.User = SampleData.CreateKaraThrace();
+
+            // Act
+            await service.InsertAsync(cylonDetector);
+            
+
+            // Assert
+            Assert.IsNotNull(cylonDetector.Id);
+        }
+
+        [TestMethod]
+        public async Task UpdateGift_SaveIntoDatabase()
+        {
+            // Arrange
+            using var dbContextInsert = new ApplicationDbContext(Options);
+            IGiftService service = new GiftService(dbContextInsert, Mapper);
+
+            var cylonDetector = SampleData.CreateCylonDetector();
```

Comment:
It is a bit difficult for me to tell that CylonDetector is a gift. I can of course figure out it that is a gift by looking at the rest of the code but since I cannot tell just by looking at the single line of code what data type CyclonDetector is, I might suggest not using var.
```suggestion
            Gift cylonDetector = SampleData.CreateCylonDetector();
```